diff --git a/Source/Visitable/VisitableView.swift b/Source/Visitable/VisitableView.swift
index 12452b5..c008c6f 100644
--- a/Source/Visitable/VisitableView.swift
+++ b/Source/Visitable/VisitableView.swift
@@ -21,6 +21,8 @@ open class VisitableView: UIView {
     open var webView: WKWebView?
     private weak var visitable: Visitable?
 
+    public var refreshControlTopAnchor: CGFloat = 0
+
     open func activateWebView(_ webView: WKWebView, forVisitable visitable: Visitable) {
         self.webView = webView
         self.visitable = visitable
@@ -76,7 +78,7 @@ open class VisitableView: UIView {
         
         NSLayoutConstraint.activate([
             refreshControl.centerXAnchor.constraint(equalTo: centerXAnchor),
-            refreshControl.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor),
+            refreshControl.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: refreshControlTopAnchor),
             refreshControl.heightAnchor.constraint(equalToConstant: refreshControlHeight)
         ])
         #endif
