diff --git a/Source/WebView/WebViewBridge.swift b/Source/WebView/WebViewBridge.swift
index a7bdc73..f0ec134 100644
--- a/Source/WebView/WebViewBridge.swift
+++ b/Source/WebView/WebViewBridge.swift
@@ -49,7 +49,7 @@ final class WebViewBridge {
     }
     
     private var userScript: WKUserScript {
-        let url = Self.bundle.url(forResource: "turbo", withExtension: "js")!
+        let url = Bundle.main.url(forResource: "turbo", withExtension: "js")!
         let source = try! String(contentsOf: url, encoding: .utf8)
         return WKUserScript(source: source, injectionTime: .atDocumentEnd, forMainFrameOnly: true)
     }
